### PR TITLE
Remove adminhtml message manager deprecated calls

### PR DESCRIPTION
**Pull request by adrian-martinez-interactiv4**
### Description
- Remove adminhtml message manager deprecated calls
- Proxy calls from deprecated methods to new ones in \Magento\Framework\Message\Manager
- Added complementary tests in \Magento\Framework\Message\ManagerTest
- Adapted unit tests, used addComplexMessages where needed to use new default renderer
- Optimized imports, phpcs-fixer for modified classes

### Fixed Issues (if relevant)
None, improvement

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@adrian-martinez-interactiv4 have you tested these changes? I had issues with escaping links in messages. It was in customer module, so all these changes should be rechecked carefully.

**Comment by adrian-martinez-interactiv4** [comment]
@ihor-sviziev yep, I was still working on that, I have had also the same problem, I've had to adapt tests and check carefully all still works as intended, but just for checking, what was the concrete problem in the customer module? How can I reproduce the issue you are talking about?

**Comment by adrian-martinez-interactiv4** [comment]
@ihor-sviziev I have found what you were referring to, I'm going to check that, thanks for pointing it out

**Comment by adrian-martinez-interactiv4** [comment]
ready to review

**Comment by ishakhsuvarov** [comment]
@adrian-martinez-interactiv4 Are you interested to provide a complimentary update for the magento-partners/magento2ee repo?
Internal CICD shows that there is some changes to be done in the EE edition as well.

**Comment by ihor-sviziev** [comment]
@adrian-martinez-interactiv4 also this PR contains conflict with 2.2-develop branch. Could you merge or rebase it?

File: app/code/Magento/Checkout/Controller/Cart/CouponPost.php [hunk 1]
```diff
@@ -95,44 +95,50 @@ public function execute()
...
                                 $escaper->escapeHtml($couponCode)
                             )
                         );
                     } else {
-                        $this->messageManager->addError(
-                            __(
-                                'The coupon code "%1" is not valid.',
-                                $escaper->escapeHtml($couponCode)
-                            )
+                        $this->messageManager->addComplexErrorMessage(
```
Comment (vkublytskyi) [inline app/code/Magento/Checkout/Controller/Cart/CouponPost.php]: @adrian-martinez-interactiv4 would you explain why error message is created with `addComplexErrorMessage` but success message with simple `addSuccessMessage `. Messaging is pretty same so in both cases should be used same approach

File: app/code/Magento/CurrencySymbol/Controller/Adminhtml/System/Currency/FetchRates.php [hunk 2]
```diff
@@ -37,18 +37,36 @@ public function execute()
...
+                        'addUnescapedMessage',
+                        [
+                            'text' => $error,
+                        ]
+                    );
                 }
-                $this->messageManager->addWarning(
-                    __('Click "Save" to apply the rates we found.')
+                $this->messageManager->addComplexWarningMessage(
+                    'addUnescapedMessage',
```
Comment (vkublytskyi) [inline app/code/Magento/CurrencySymbol/Controller/Adminhtml/System/Currency/FetchRates.php]: What is the reason to use unescaped Message here? Seems message should not contain any HTML markup.

File: app/code/Magento/Backend/Controller/Adminhtml/Cache/MassDisable.php [hunk 3]
```diff
@@ -60,12 +60,17 @@ private function disableCache()
             }
             if ($updatedTypes > 0) {
                 $this->_cacheState->persist();
-                $this->messageManager->addSuccess(__("%1 cache type(s) disabled.", $updatedTypes));
+                $this->messageManager->addSuccessMessage(__("%1 cache type(s) disabled.", $updatedTypes));
             }
         } catch (LocalizedException $e) {
-            $this->messageManager->addError($e->getMessage());
+            $this->messageManager->addComplexErrorMessage(
+                'addUnescapedMessage',
```
Comment (vkublytskyi) [inline app/code/Magento/Backend/Controller/Adminhtml/Cache/MassDisable.php]: @adrian-martinez-interactiv4 Where is `addUnescapedMessage` message renderer defined? I can not find it in `2.2-develop` or this PR. Before use of complex message it should be registered in `\Magento\Framework\View\Element\Message\MessageConfigurationsPool` (e.g. https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/UrlRewrite/etc/adminhtml/di.xml#L19)

**Comment by adrian-martinez-interactiv4** [comment]
@ihor-sviziev @vkublytskyi I'm back from my holidays, I'm checking all your comments

**Comment by ihor-sviziev** [comment]
Hi @adrian-martinez-interactiv4 any updates? Will you continue working on this PR?

File: app/code/Magento/Checkout/Controller/Cart/CouponPost.php (hunk 1 above)
Comment (adrian-martinez-interactiv4) [reply app/code/Magento/Checkout/Controller/Cart/CouponPost.php]: Yes, it seems so, it seems to be a mistake, but I don't remember if I did it this for some reason, I'm checking it

File: app/code/Magento/CurrencySymbol/Controller/Adminhtml/System/Currency/FetchRates.php (hunk 2 above)
Comment (adrian-martinez-interactiv4) [reply app/code/Magento/CurrencySymbol/Controller/Adminhtml/System/Currency/FetchRates.php]: I'll check this, I can't remember if it was about double quotes or if it is a mistake

**Comment by adrian-martinez-interactiv4** [comment]
@ihor-sviziev Yes, I have some open PRs and forgot to review this one, sorry

**Comment by ihor-sviziev** [comment]
@adrian-martinez-interactiv4  integration tests are failed, please review build on travis

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please adjust integration tests

**Comment by adrian-martinez-interactiv4** [comment]
@ihor-sviziev integration tests fixed, codacy is complaining about an existing variable length

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
Hi @adrian-martinez-interactiv4,
As we discussed in Slack - please fix conflicts in this PR and I'll review them again.

**Comment by ihor-sviziev** [comment]
Hi @adrian-martinez-interactiv4,
I'm closing this PR due to inactivity. Feel free to re-open or create new PR once you'll resolve conflicts. You can contact me in Slack or just in PR comments if you have any questions.

**Pull request closed without merging**